{{cookiecutter.project_dir}}/Jenkinsfile

```diff
@@ -0,0 +1,32 @@
+pipeline {
+    agent {
+        label 'master'
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '5'))
+    }
+    parameters {
+        string(name: 'BRANCH', defaultValue: 'master')
+        string(name: 'DOCKER_IMAGE_TAG', defaultValue: 'latest', description: 'Image to be deployed')
+        booleanParam(name: 'SETUP', defaultValue: false, description: 'Check the box if Ansible roles tagged as `setup` should be ran too.')
+    }
+    stages {
+        stage('Deploy') {
+            steps {
+                script {
+                    if (params.SETUP) {
+                        env.TAGS="setup,deploy"
+                    } else {
+                        env.TAGS="deploy"
+                    }
+                }
+                sh "ansible-playbook -i ansible/inventory.ini -u jenkins --tags \"${env.TAGS}\" -e \"docker_image_tag=${params.DOCKER_IMAGE_TAG}\" --ssh-extra-args=\"-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o GlobalKnownHostsFile=/dev/null\" ansible/deploy.yml"
```

Comment:
[miękka dywagacja]
Hm, to jest Jenkinsfile odpowiedni dla repo deployującego zdalną maszynę, tak?
Na pewno chcemy mieć takie coś jako 'default'?

Reply:
IMO tak, łatwiej coś usunąć niż kopiować z innych projektów.

Reply:
Skoro to Cookiecutter, to możesz zrobić z tej pozycji linię warunkową (bo to przecież wszystko Jinja jest). Dodasz tylko dodatkowe pytanie o Ansible'a przy odpalaniu cookiecuttera i voilà :)

Tak samo można to ograć z tym GCP'em.